pkg/lib/authn/otp/service.go

```diff
@@ -146,13 +170,20 @@ func (s *Service) SetUserInputtedCode(target string, userInputtedCode string) (*
 }
 
 func (s *Service) SetUserInputtedMagicLinkCode(userInputtedCode string) (*Code, error) {
-	codeModel, err := s.getCode(userInputtedCode)
+	target, err := s.MagicLinkStore.Get(userInputtedCode)
```

Comment:
If it is not found, do we need to return `ErrInvalidMagicLink` instead? Please ensure the UI shows the current error if the code is invalid.